src/api/app/models/staging/stage_requests.rb

```diff
@@ -117,8 +121,45 @@ def add_review_for_staged_request(request)
     request.change_review_state('accepted', by_group: staging_workflow.managers_group.title, comment: "Picked \"#{staging_project}\"")
   end
 
-  def add_review_for_unstaged_request(request)
+  def add_review_for_unstaged_request(request, staging_project)
     request.addreview(by_group: staging_workflow.managers_group.title, comment: "Being evaluated by group \"#{staging_workflow.managers_group}\"")
     request.change_review_state('accepted', by_project: staging_project.name, comment: "Moved back to project \"#{staging_workflow.project}\"")
   end
+
+  def remove_packages(request, staging_project)
+    package_names = request.bs_request_actions.pluck(:target_package)
+    staging_project_packages = staging_project.packages.where(name: package_names)
+    staging_project_packages_names = staging_project_packages.pluck(:name)
+    deleted_packages = staging_project_packages.destroy_all.pluck(:name)
+
+    return staging_project_packages_names - deleted_packages
+  end
+
+  def unstageable?(request, staging_project)
+    return true if staging_project && staging_project.overall_state != :accepting
+    errors << if staging_project.nil?
+                "Request '#{request.number}' is not staged"
+              else
+                "Can't change staged requests '#{request.number}': Project '#{staging_project}' is being accepted."
+              end
+    return false
+  end
+
+  def missing_packages(not_deleted_packages)
+    return if not_deleted_packages.empty?
+
+    message = not_deleted_packages.map { |staging_project, package_names| "'#{package_names.flatten.to_sentence}' from '#{staging_project}'" }
+    errors << "The following packages couldn't be deleted: #{message.to_sentence}"
```

Comment:
Hm under which conditions can this happen?

Reply:
I just wanted to be sure that it was controlled if something strange happens. Are you suggesting that this is not needed?

Reply:
No I want the error message to explain what is going on and what people can do about it.